pkg/operator/petset.go

```diff
@@ -126,19 +122,31 @@ func makePetSetService(p *spec.Prometheus) *v1.Service {
 	return svc
 }
 
-func makePetSetSpec(name, image, version, retention string, replicas int32, alertmanagers []string) v1alpha1.PetSetSpec {
+func makePetSetSpec(p spec.Prometheus, alertmanagers []string) v1alpha1.PetSetSpec {
 	// Prometheus may take quite long to shut down to checkpoint existing data.
 	// Allow up to 10 minutes for clean termination.
 	terminationGracePeriod := int64(600)
 
+	// We attempt to specify decent storage tuning flags based on how much the
+	// requested memory can fit. The user has to specify an appropriate buffering
+	// in memory limits to catch increased memory usage during query bursts.
+	// More info: https://prometheus.io/docs/operating/storage/.
+	reqMem := p.Spec.Resources.Requests[v1.ResourceMemory]
+	// 1024 byte is the fixed chunk size. With increasing number of chunks actually
+	// in memory, overhead owed to their management, higher ingestion buffers, etc.
+	// increases.
+	// We are conservative for now an assume this to be 80% as the Kubernetes environment
+	// generally has a very high time series churn.
+	memChunks := reqMem.Value() / 1024 / 5
```

Comment:
Yea, see prometheus/prometheus#455 for long-running tracking discussion.
Lot's of things not really understood about this as there are so many factors. Especially the time series churn in Kubernetes.

20% seems safe for now. 30% is the go-to number at the moment.
Obviously the data held in memory should be more at the order of 80% – hopefully we can address this upstream at some point.